### PR TITLE
Update the Gubernator README with GCS layout

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This is currently just tribal knowledge.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @fejta @kargakis @rmmh 
/assign @rmmh

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
```
$ gubernator/verify_config.sh 
Gubernator configuration file is out of sync!
31d30
<   - pull-kubernetes-federation-e2e-gce
Run gubernator/update-config.sh
```

Did someone merge over broken tests?

**Comment by stevekuznetsov** [comment]
Oh, we should trigger than when someone touches the Prow config as well as Gubernator

File: gubernator/README.md [hunk 1]
```diff
@@ -29,3 +29,71 @@ For deployment:
...
+}
+```
+
+The following fields in `finished.json` are honored:
+
+```json
+{
+    "timestamp": "seconds after UNIX epoch that the build finished",
+    "result": "human-readable string detailing the result of the build",
+    "passed": "boolean, whether the build succeeded",
```
Comment (rmmh) [inline gubernator/README.md]: nothing uses this

File: gubernator/README.md [hunk 2]
```diff
@@ -29,3 +29,71 @@ For deployment:
...
+    },
+}
+```
+
+The following fields in `finished.json` are honored:
+
+```json
+{
+    "timestamp": "seconds after UNIX epoch that the build finished",
+    "result": "human-readable string detailing the result of the build",
```
Comment (rmmh) [inline gubernator/README.md]: build result (SUCCESS or FAILED)

File: gubernator/README.md [hunk 3]
```diff
@@ -29,3 +29,71 @@ For deployment:
...
+
+# GCS Layout API
+
+In order to correctly interpret jobs results, in GCS, Gubernator expects that
+any one job directory is laid out in a specific manner, and that job directories
+are laid out in a specific manner relative to each other.
+
+## Job Artifact GCS Layout
+
+For a single build of a job, Gubernator expects the following layout in GCS:
```
Comment (rmmh) [inline gubernator/README.md]: Maybe copy the line from docs/federated_testing.md -- "Test results for a build are stored on GCS as gs://$PREFIX/$JOB_NAME/$BUILD_NUMBER. Every run should upload started.json, finished.json, and build-log.txt, and can optionally upload JUnit XML to the artifacts/ directory."

File: gubernator/README.md [hunk 4]
```diff
@@ -29,3 +29,71 @@ For deployment:
...
+```
+
+The following fields in `finished.json` are honored:
+
+```json
+{
+    "timestamp": "seconds after UNIX epoch that the build finished",
+    "result": "human-readable string detailing the result of the build",
+    "passed": "boolean, whether the build succeeded",
+}
```
Comment (rmmh) [inline gubernator/README.md]: `"metadata": "dictionary of additional key-value pairs that will be displayed to the user"`

File: gubernator/README.md [hunk 5]
```diff
@@ -29,3 +29,71 @@ For deployment:
         }
     }
 ```
+
+# GCS Layout API
```
Comment (rmmh) [inline gubernator/README.md]: drop "API"

**Comment by rmmh** [comment]
Thanks for writing this up! Can you link to this from docs/federated_testing.md as well?

**Comment by stevekuznetsov** [comment]
@rmmh updated, added the link to the federated testing doc

**Pull request merged**